phd-tests/framework/src/artifacts/buildomat.rs

```diff
@@ -35,48 +35,50 @@ impl Repo {
         Self(Cow::Borrowed(s))
     }
 
-    pub(super) fn artifact_for_commit(
+    pub(super) async fn artifact_for_commit(
         self,
         series: Series,
         commit: Commit,
         filename: impl AsRef<Utf8Path>,
         downloader: &DownloadConfig,
     ) -> anyhow::Result<BuildomatArtifact> {
         let filename = filename.as_ref();
-        let sha256 = self.get_sha256(&series, &commit, filename, downloader)?;
+        let sha256 =
+            self.get_sha256(&series, &commit, filename, downloader).await?;
 
         Ok(BuildomatArtifact { repo: self, series, commit, sha256 })
     }
 
-    pub(super) fn get_branch_head(
+    pub(super) async fn get_branch_head(
         &self,
         branch: &str,
     ) -> anyhow::Result<Commit> {
-        (|| {
+        (|| async {
             let uri = format!("{BASE_URI}/branch/{self}/{branch}");
-            let client = reqwest::blocking::ClientBuilder::new()
+            let client = reqwest::ClientBuilder::new()
```

Comment:
this isn't directly related, so we should probably save it for a future PR, but a quick note is that we may want to eventually have one persistent `reqwest::Client` that the artifact store owns and uses for each buildomat artifact. I believe the `reqwest::Client` owns a TCP connection pool, so reusing a client might let us avoid establishing a new  TCP connection every time we try to download something, shaving a little time off downloads...